-1

Tested at: b21f5a30fc8fe8b5d2cbbc778bbb07fbf5d8897f

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-17627/17963/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>In file included from /cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/boost/1.57.0-ogepgl/include/boost/lexical_cast/detail/converter_lexical_streams.hpp:64:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/boost/1.57.0-ogepgl/include/boost/lexical_cast/detail/inf_nan.hpp:34:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/boost/1.57.0-ogepgl/include/boost/math/special_functions/sign.hpp:16:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/boost/1.57.0-ogepgl/include/boost/math/tools/config.hpp:335:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/boost/1.57.0-ogepgl/include/boost/detail/fenv.hpp:97:
/cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/fenv.h:58:11: error: no member named 'fenv_t' in the global namespace
  using ::fenv_t;
        ~~^
/cvmfs/cms-ib.cern.ch/nweek-02460/slc6_amd64_gcc530/external/gcc/5.3.0/include/c++/5.3.0/fenv.h:59:11: error: no member named 'fexcept_t' in the global namespace
  using ::fexcept_t;
        ~~^
</pre>

--- a/Geometry/CaloTopology/src/CaloTowerConstituentsMap.cc
+++ b/Geometry/CaloTopology/src/CaloTowerConstituentsMap.cc
@@ -97,8 +97,16 @@ std::vector<DetId> CaloTowerConstituentsMap::constituentsOf(const CaloTowerDetId
     if (id.ietaAbs()<=m_cttopo->lastHBRing()) {
       m_hcaltopo->depthBinInformation(HcalBarrel,hcal_ieta,id.iphi(),id.zside(),nd,sd);
       for (int i=0; i<nd; i++) {
-	HcalDetId hid = m_hcaltopo->mergedDepthDetId(HcalDetId(HcalBarrel,hcal_ieta*id.zside(),id.iphi(),i+sd));
-        if (std::find(items.begin(),items.end(),hid) == items.end()) {
+	if (m_hcaltopo->withSpecialRBXHBHE()) {
+	  HcalDetId hid = m_hcaltopo->mergedDepthDetId(HcalDetId(HcalBarrel,hcal_ieta*id.zside(),id.iphi(),i+sd));
+	  if (std::find(items.begin(),items.end(),hid) == items.end()) {
+	    items.push_back(hid);
+#ifdef EDM_ML_DEBUG
+	    std::cout << id << " Depth " << i << ":" << i+sd << " " << hid <<"\n";
+#endif
+	  }
+	} else {
+	  HcalDetId hid(HcalBarrel,hcal_ieta*id.zside(),id.iphi(),i+sd);
 	  items.push_back(hid);
 #ifdef EDM_ML_DEBUG
 	  std::cout << id << " Depth " << i << ":" << i+sd << " " << hid <<"\n";
@@ -123,8 +131,16 @@ std::vector<DetId> CaloTowerConstituentsMap::constituentsOf(const CaloTowerDetId
     if (id.ietaAbs()>=m_cttopo->firstHERing() && id.ietaAbs()<=m_cttopo->lastHERing()) {
       m_hcaltopo->depthBinInformation(HcalEndcap,hcal_ieta,id.iphi(),id.zside(),nd,sd);
       for (int i=0; i<nd; i++) {
-	HcalDetId hid = m_hcaltopo->mergedDepthDetId(HcalDetId(HcalEndcap,hcal_ieta*id.zside(),id.iphi(),i+sd));
-        if (std::find(items.begin(),items.end(),hid) == items.end()) {
+	if (m_hcaltopo->withSpecialRBXHBHE()) {
+	  HcalDetId hid = m_hcaltopo->mergedDepthDetId(HcalDetId(HcalEndcap,hcal_ieta*id.zside(),id.iphi(),i+sd));
+	  if (std::find(items.begin(),items.end(),hid) == items.end()) {
+	    items.push_back(hid);
+#ifdef EDM_ML_DEBUG
+	    std::cout << id << " Depth " << i << ":" << i+sd << " " << hid <<"\n";
+#endif
+	  } 
+	} else {
+	  HcalDetId hid(HcalEndcap,hcal_ieta*id.zside(),id.iphi(),i+sd);
 	  items.push_back(hid);
 #ifdef EDM_ML_DEBUG
 	  std::cout << id << " Depth " << i << ":" << i+sd << " " << hid <<"\n";

